src/SerilogWeb.Classic/Classic/ApplicationLifecycleModule.cs

```diff
@@ -155,7 +167,7 @@ public void Init(HttpApplication application)
                     stopwatch.Stop();
 
                     var request = HttpContextCurrent.Request;
-                    if (request == null)
+                    if (request == null || !(_requestPredicates ?? Enumerable.Empty<Func<HttpContext, bool>>()).All(p => p(application.Context)))
```

Comment:
Could we try:

```csharp
if (request == null || _requestPredicates.Count != 0 && _requestPredicates.Any(p => !p(application.Context)))
```

Saves a closure allocation/iteration if no predicates are configured.